workspace_tools/targets.py

```diff
@@ -1633,6 +1760,13 @@ def __init__(self):
         self.supported_toolchains = ["ARM", "GCC_ARM", "IAR"]
         self.supported_form_factors = ["ARDUINO"]
         self.default_toolchain = "ARM"
+        self.progen = {
```

Comment:
This dict needs documentation ! I think we should create readme in exporters, where this is documented what progen expects. As we can see , this can be expanded by flags for various tools. or in the future other data which might be required - as stated in the commit message